docs/prompting/zero_shot/role_prompting.md

```diff
@@ -1,7 +1,65 @@
 ---
-title: ""
+title: "Role Prompting"
 description: ""
-keywords: ""
+keywords: "prompt, prompting, prompt engineering, llm, gpt, openai, role prompting, persona prompting, model, AI, python, instructor, zero-shot"
 ---
 
-[wip]
+Role prompting<sup><a href="https://arxiv.org/abs/2310.00746">1</a>, <a href="https://arxiv.org/abs/2311.10054">2</a></sup>, or persona prompting<sup><a href="#ref-3">3</a>, <a href="https://arxiv.org/abs/2307.05300">4</a></sup>, assigns a specific role to the model in the prompt ("You are ..."). This can improve outputs for open-ended tasks<sup><a href="https://dl.acm.org/doi/10.1145/3411763.3451760">5</a></sup> and improve accuracy on benchmarks<sup><a href="https://arxiv.org/abs/2311.10054">2</a></sup>.<sup><a href="https://arxiv.org/abs/2406.06608">\*</a></sup>
+
+```python
+import openai
+import instructor
+
+from typing import List, Literal
+from pydantic import BaseModel, Field
+
+# Apply the patch to the OpenAI client
+# enables response_model keyword
+client = instructor.from_openai(openai.OpenAI())
+
+LABELS = Literal["ACCOUNT", "BILLING", "GENERAL_QUERY"]
+
+
+class MultiClassPrediction(BaseModel):
+    labels: List[LABELS] = Field(
+        ...,
+        description="Only select the labels that apply to the support ticket.",
+    )
+
+
+def multi_classify(data: str) -> MultiClassPrediction:
+    return client.chat.completions.create(
+        model="gpt-4o",
+        response_model=MultiClassPrediction,
+        messages=[
+            {
+                "role": "system",
+                "content": f"You are a support agent at a tech company. Only select the labels that apply to the support ticket.",
+            },
+            {
+                "role": "user",
+                "content": f"Classify the following support ticket: {data}",
+            },
+        ],
+    )  # type: ignore
+
+
+if __name__ == "__main__":
```

Comment:
Remove if name == main, just run the script. Don't use the assert, just use a # <what is printed out>